internal/pkg/connectors/currencycloud/config.go

```diff
@@ -0,0 +1,69 @@
+package currencycloud
+
+import (
+	"encoding/json"
+	"time"
+)
+
+type Config struct {
+	LoginID       string   `json:"loginID" bson:"loginID"`
+	APIKey        string   `json:"apiKey" bson:"apiKey"`
+	Endpoint      string   `json:"endpoint" bson:"endpoint"`
+	PollingPeriod Duration `json:"pollingPeriod" bson:"pollingPeriod"`
+}
+
+func (c Config) Validate() error {
+	if c.APIKey == "" {
+		return ErrMissingAPIKey
+	}
+
+	if c.LoginID == "" {
+		return ErrMissingLoginID
+	}
+
+	if c.PollingPeriod == 0 {
+		return ErrMissingPollingPeriod
+	}
+
+	return nil
+}
+
+type Duration time.Duration
```

Comment:
A previous PR (merged) has added a custom time.Duration.
I prefer this implementation.
Can you rebase and update?
@antoinegelloz

Reply:
The PR that you are talking about @gfyrag has indeed been merged, but not yet in main. It was based on this feat branch (https://github.com/formancehq/payments/pull/50/files) that still needs to be reviewed.